lab3/client/src/views/button.js

```diff
@@ -0,0 +1,29 @@
+import constants from '../constants'; // 
+
+export default function (store) {
+	return class ButtonComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+
+			this.onStateChange = this.handleStateChange.bind(this);
+
+			// TODO: add click event to increment counter
+			// hint: use "store.dispatch" method (see example component)
+
+		}
+
+		handleStateChange (newState) {
+			this.innerHTML = `Strawberries: ${newState.counter}`;
+		}
+		
+		// InnerHTML rendering, this code executes once when you create the object in HTML 
+		connectedCallback () {
+			this.innerHTML = '<button>Click me</button>';
```

Comment:
I'd recommend you to finish the implementation (copying from the example I had in class) to do the button increment function so that it's easier to do end to end testing. (e.g. clicking on the button multiple times to buy the first generator).

Reply:
Finished!